ddtrace/internal/processor/trace.py

```diff
@@ -68,7 +69,11 @@ def process_trace(self, trace):
             if self._compute_stats_enabled:
                 priority = trace[0]._context.sampling_priority if trace[0]._context is not None else None
                 if priority is not None and priority <= 0:
-                    return None
+                    # When any span is marked as keep by a single span sampling
+                    # decision then we still send all and only those spans.
+                    single_spans = [_ for _ in trace if is_single_span_sampled(_)]
+
+                    return single_spans or None
```

Comment:
Is there a difference between returning None or an empty list?

Reply:
yes it needs to be `None` since there is an explicit check for `if spans is None` when calling trace processors.